pkg/types/validation/installconfig.go

```diff
@@ -418,34 +418,34 @@ func validateClusterNetwork(n *types.Networking, cn *types.ClusterNetworkEntry,
 	return allErrs
 }
 
-func validateControlPlane(platform *types.Platform, pool *types.MachinePool, fldPath *field.Path) field.ErrorList {
+func validateControlPlane(featureSet configv1.FeatureSet, platform *types.Platform, pool *types.MachinePool, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 	if pool.Name != types.MachinePoolControlPlaneRoleName {
 		allErrs = append(allErrs, field.NotSupported(fldPath.Child("name"), pool.Name, []string{types.MachinePoolControlPlaneRoleName}))
 	}
 	if pool.Replicas != nil && *pool.Replicas == 0 {
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("replicas"), pool.Replicas, "number of control plane replicas must be positive"))
 	}
-	allErrs = append(allErrs, ValidateMachinePool(platform, pool, fldPath)...)
+	allErrs = append(allErrs, ValidateMachinePool(featureSet, platform, pool, fldPath)...)
 	return allErrs
 }
 
-func validateCompute(platform *types.Platform, control *types.MachinePool, pools []types.MachinePool, fldPath *field.Path) field.ErrorList {
+func validateCompute(featureSet configv1.FeatureSet, platform *types.Platform, control *types.MachinePool, pools []types.MachinePool, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 	poolNames := map[string]bool{}
-	for i, p := range pools {
```

Comment:
This doesn't seem crucial and probably would have been better left to a separate commit/PR to minimise the diff

Reply:
The linter didn’t let me commit without this change. The static analysis on this file was probably triggered by my change, which appears, as you note, unrelated.

Reply:
It could have been a separate commit indeed.

Reply:
It's a "feature" of the new linter: it only analyses changes after a specific revision (`--new-from-rev=75173a17cf`) so we don't have to fix all the warnings in the entire codebase at once. The downside is that "old" warnings are triggered when new changes touch the area. A separate commit is a good compromise here, otherwise you'd have to wait for the linting fix PR to be merged.